### PR TITLE
History, Python samples

### PR DESCRIPTION
**Pull request by bartlomiejwereszczynski**

### CODE REVIEW

**Comment by mattheworiordan** [comment]
Hi @bartlomiejwereszczynski thanks. Before I review, is there an associated docs PR that has the tutorial instructions I can review at the same time?

**Comment by bartlomiejwereszczynski** [comment]
Hi @mattheworiordan , Yesy but since i had to start from `tutorials-in-php` (needed to change some blocks) i wanted to wait until you'll check them. You can view current progress in https://github.com/ably/docs/pull/164

File: history.py [hunk 1]
```diff
@@ -0,0 +1,4 @@
+from __future__ import unicode_literals
+from ably import AblyRest
+
+client = AblyRest('{{ApiKey}}')
```
Comment (mattheworiordan) [inline history.py:4]: This is missing the friendly error message for users who run this demo without setting the API Key.  It is important that the user is given an error message that makes it obvious they need to add the API to run the demo

File: history.py [hunk 2]
```diff
@@ -2,3 +2,13 @@
 from ably import AblyRest
 
 client = AblyRest('{{ApiKey}}')
+
+print('Retrieving channel...')
```
Comment (mattheworiordan) [inline history.py:6]: As with PHP, I don't think we should state retrieving a channel here.  We are instancing a stateless channel so not sure this adds anything but may in fact confuse users by making them think some server-side operation is done when instancing a REST channel.

File: history.py [hunk 3]
```diff
@@ -12,3 +12,19 @@
 channel.publish('play', 'bark')
 channel.publish('play', 'meow')
 channel.publish('play', 'cluck')
+
+print('Releasing channel...')
+# this part is only for tutorial purposes, in real you can use same instance of $channel
```
Comment (mattheworiordan) [inline history.py:17]: We should remove this, but you are referencing the PHP variable $channel here anyway

**Review by mattheworiordan — CHANGES_REQUESTED** [review]
Thanks, looks good, just a few bits of feedback

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1 @@
+.idea
```
Comment (mattheworiordan) [inline .gitignore:1]: Isn't this ignore specific to your editor you are using?  If so, that should not be added to this repo and instead you should perhaps add to your global git ignore

**Comment by bartlomiejwereszczynski** [comment]
recreated version is available here: https://github.com/ably/tutorials/pull/10

**Comment by mattheworiordan** [comment]
@bartlomiejwereszczynski why is this up as another PR?  I would really prefer to review what you have fixed rather than review everything all over again?

**Pull request closed without merging**